### ISSUE TITLE
Access to Rate Limit headers 

### ISSUE THREAD

**Issue by harlow** [ChimeraCoder/anaconda#77]
Hello, I noticed the `ApiError` has access to the response headers. However, I was wondering if there is a way to access the headers on a success response too?

On any given successful response there is also rate-limit information passed back in the headers.
https://dev.twitter.com/rest/public/rate-limiting

```
X-Rate-Limit-Limit: the rate limit ceiling for that given request
X-Rate-Limit-Remaining: the number of requests left for the 15 minute window
X-Rate-Limit-Reset: the remaining window before the rate limit resets in UTC epoch seconds
```

If this is currently not available let me know if you have thoughts on where this could fit in the library. I'd be happy to help build it in.

**Comment by ChimeraCoder** [comment]
Yeah, I've actually been thinking about adding (optionally) better logging in general. The trick is that we don't want to pollute people's logs when they don't want this, so it has to be optional. The straightforward way is to have a logger that only outputs values if a certain flag is set. This is an interface that's pretty easy to implement wrapping the stdlib's logger.

Another option is to use a channel that can feed on logging values emitted by the API client. This is similar to what Go lexers (including the `template` package, IIRC) do. There are some advantages to doing this, but it may be overkill here, especially if this is the only place that this sort of logging is needed.

Could you elaborate a bit more on the use case here (the specific problem you're running into, in which the headers would be helpful)? That may help us to figure out the best way to add this while maintaining the same interface for API calls that people are used to.

**Comment by harlow** [comment]
OK sounds good. Thanks for the quick reply.

> Could you elaborate a bit more on the use case here (the specific problem you're running into, in which the headers would be helpful)? 

I was thinking it would be nice to have access to the rate limit information (for successful queries too). As I think about it some more though the Throttling functionality should suffice, and then we could log when we hit our limits.

**Comment by ChimeraCoder** [comment]
@harlow This is fixed, as of bdb4b76. I didn't realize how simple it would be; I forgot we already had the logger from the streaming API (which is separate on Twitter's end, but included in this library).

Let me know if this satisfies what you're looking for - at this point, it'd be easy to tweak it as needed.
